src/util/scroll.js

```diff
@@ -3,63 +3,64 @@ import { changeCenter } from './path.js';
 
 export const actions = {
     exit: (response) => {
-        response.element.classList.remove('active');
+        const { element } = response;
+        element.classList.remove('active');
     },
     enter: (response, config, map) => {
-        const { element } = response;
-        element.classList.add('active');    
-        
-        const chapter = config.chapters.find(
-            (chap) => chap.id === response.index
-        );
-        const speciesList = element.querySelectorAll('.icon');;
-    
-        Array.from(speciesList).forEach( (item, idx) => {
+        const { index, element } = response;
+        element.classList.add('active');
+
+        [...element.querySelectorAll('.icon')].forEach((item, idx) => {
             item.style.animationTimingFunction = "ease-in-out";
             item.style.animationDelay = `${idx * 0.1}s`;
         });
 
         if (element.className.includes('chapter')) {
-            map.getSource('pointSource').setData(point(chapter.location.center));
-            map.getSource('pointSourceBg').setData(point(chapter.location.center));
+            const chapter = config.chapters.find((chap) => chap.id === index);
+            const center = point(chapter.location.center)
+            map.getSource('pointSource').setData(center);
+            map.getSource('pointSourceBg').setData(center);
         }
-   
     },
     progress: (response, config, map, data) => {
-
         const { index, progress, element } = response;
-        let stepProgress;
-        let movingLine;
-    
-        const chapter = config.chapters.find(
-            (chap) => chap.id === response.index
-        );
-    
-    
+        const chapter = config.chapters.find(chapter => chapter.id === index);
+
+        const atBottom = element.className.includes('footer')
+            || document.body.offsetHeight <= window.innerHeight + window.scrollY;
+
         if (element.className.includes('header')) {
-            response.element.style.opacity = Math.round(1 - response.progress);
+            element.style.opacity = Math.round(1 - progress);
+        }
+
+        const setCenter = center => {
+            if (!config.followPoint || !center || atBottom) return;
+
+            const { lng, lat } = map.getCenter()
```

Comment:
this is a neat! It seems like you use some rounding threshold between current map center and path coordinate to define whether the map should be recentered. One odd thing I'm noticing is that when backscrolling from `footer` to the last `chapter`, `map.getCenter()` coordinates fluctuate, even though the coordinates of the path didn't change, which causes the map to recenter. Not quite sure why this is happening tho

Reply:
I've noticed this problem and have a few approaches that can fix it.

It's caused by the fact that `easeTo` function of the map applies a slight animation with zoom (which causes the center to move slightly) even if you're already positioned where you wanted, and if this animation is slow enough, it can cause an infinite loop because the animation to start playing, then the check is performed too early and sees that map is in the wrong position, causing the animation to start again.

Reply:
One of the simplest approaches to this fix would be instead of comparing the current state of the map with where we want it to be, to save the last requested target of `easeTo` somewhere and skip the next call to it if the target is the same as the last time

Reply:
Added a patch to fix the issue, hope I correctly understood your description)